mreg/tests.py

```diff
@@ -0,0 +1,67 @@
+from django.test import TestCase
+
+from mreg.models import (Cname, HinfoPreset, Host, Ipaddress, NameServer,
+                         Naptr, PtrOverride, Srv, Network, Txt, ForwardZone,
+                         ReverseZone, ModelChangeLog)
+
+
+def clean_and_save(entity):
+    entity.full_clean()
+    entity.save()
+
+
+class NameServerDeletionTestCase(TestCase):
+    """This class defines the test suite for the NameServer model."""
+
+    def setUp(self):
+        """Define the test client and other test variables."""
+        self.zone_sample = ForwardZone(name='example.org',
+                                       primary_ns='ns.example.org',
+                                       email='hostmaster@example.org')
+        self.zone_1010 = ReverseZone(name='127.in-addr.arpa',
+                                     primary_ns='ns.example.org',
+                                     email='hostmaster@example.org')
+        clean_and_save(self.zone_sample)
+        clean_and_save(self.zone_1010)
+
+        self.network_sample = Network(range='129.240.202.0/20',
+                                      description='some description',
+                                      vlan=123,
+                                      dns_delegated=False)
+        self.ns_hostsample = Host(name='ns.example.org',
+                             contact='mail@example.org')
+        clean_and_save(self.ns_hostsample)
```

Comment:
You forgot to add the nameserver as a nameserver for the zone(s).
Something like:
```
self.zone_sample.nameservers.add(self.ns_sample)
self.zone_sample.save()
```